packages/sdk/src/conversation/middlewares/cardActionMiddleware.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ActivityTypes,
+  CardFactory,
+  InvokeResponse,
+  MessageFactory,
+  Middleware,
+  StatusCodes,
+  TurnContext,
+} from "botbuilder";
+import { AdaptiveCardResponse, TeamsFxAdaptiveCardActionHandler } from "../interface";
+import { InvokeResponseFactory } from "../invokeResponseFactory";
+
+/**
+ * @internal
+ */
+export class CardActionMiddleware implements Middleware {
+  public readonly actionHandlers: TeamsFxAdaptiveCardActionHandler[] = [];
+  private readonly defaultMessage: string = "Your response was sent to the app";
+
+  constructor(handlers?: TeamsFxAdaptiveCardActionHandler[]) {
+    if (handlers && handlers.length > 0) {
+      this.actionHandlers.push(...handlers);
+    }
+  }
+
+  async onTurn(context: TurnContext, next: () => Promise<void>): Promise<void> {
+    if (context.activity.name === "adaptiveCard/action") {
+      const action = context.activity.value.action;
+      const actionVerb = action.verb;
+
+      for (const handler of this.actionHandlers) {
+        if (handler.triggerVerb === actionVerb) {
```

Comment:
What if two handlers define the same verb?

Reply:
For each adaptive card invoke action, the bot should send an [InvokeResponse](https://docs.microsoft.com/en-us/adaptive-cards/authoring-cards/universal-action-model#response-format) to respond to the invoke request. If multiple invoke responses were sent, only the first invoke response will be taken in the underlying bot framework.

So even we don't have a `break` here, only the invoke response sent by the first matched handler will be taken, while the response sent from other handlers will be ignored in this case. 

But since this is hidden logic from Bot Framework, I will add `break` here to make it more clear and stop processing other handlers although it will be ignored if matched handler is invoked.